bot/exts/moderation/infraction/management.py

```diff
@@ -201,29 +201,26 @@ async def infraction_edit(
     # region: Search infractions
 
     @infraction_group.group(name="search", aliases=('s',), invoke_without_command=True)
-    async def infraction_search_group(self, ctx: Context, query: t.Union[UserMention, Snowflake, str]) -> None:
+    async def infraction_search_group(self, ctx: Context, query: t.Union[MemberOrUser, Snowflake, str]) -> None:
         """Searches for infractions in the database."""
         if isinstance(query, int):
             await self.search_user(ctx, discord.Object(query))
-        else:
+        elif isinstance(query, str):
             await self.search_reason(ctx, query)
+        else:
+            await self.search_user(ctx, query)
 
     @infraction_search_group.command(name="user", aliases=("member", "id"))
-    async def search_user(self, ctx: Context, user: t.Union[discord.User, proxy_user]) -> None:
+    async def search_user(self, ctx: Context, user: t.Union[MemberOrUser, discord.Object]) -> None:
```

Comment:
Shouldn't this use the `Snowflake` converter too?

Reply:
It needed to be `discord.Object` because that's what I passed in line 207.

Either I keep this as `discord.Object`, or I make it so that L207 passes the int and have the type annotation as `t.Union[MemberOrUser, t.Union[Snowflake, int]]`. Which of these 2 do you prefer?

Reply:
#1731 would fix annotations of converters. Anyway, I suppose the extra validation from `Snowflake` isn't really necessary. I think the group command uses it to disambiguate reasons that look like IDs.